File: CSharp/Selenium.WebTests/Pages/CheckoutPage.cs
using OpenQA.Selenium;

namespace Selenium.WebTests.Pages;

public class CheckoutPage : BasePage
{
    private static readonly By FirstNameInput  = By.Id("first-name");
    private static readonly By LastNameInput   = By.Id("last-name");
    private static readonly By PostalCodeInput = By.Id("postal-code");
    private static readonly By ContinueButton  = By.Id("continue");
    private static readonly By FinishButton    = By.Id("finish");
    private static readonly By ConfirmHeader   = By.ClassName("complete-header");

    public CheckoutPage(IWebDriver driver) : base(driver) { }

    public CheckoutPage FillShippingInfo(string firstName, string lastName, string postalCode)
    {
        WaitForElement(FirstNameInput).SendKeys(firstName);
        WaitForElement(LastNameInput).SendKeys(lastName);
        WaitForElement(PostalCodeInput).SendKeys(postalCode);
        ClickUntil(ContinueButton, d => d.Url.Contains("checkout-step-two"));
        return this;
    }

    public CheckoutPage FinishOrder()
    {
        ClickUntil(FinishButton, d => d.Url.Contains("checkout-complete"));
        return this;
    }

    public bool IsOrderConfirmed =>
        IsElementVisible(ConfirmHeader) &&
        Driver.FindElement(ConfirmHeader).Text.Contains("Thank you");
}

File: Java/playwright-tests/src/main/java/com/qa/portfolio/pages/CheckoutPage.java
package com.qa.portfolio.pages;

import com.microsoft.playwright.Page;

public class CheckoutPage extends BasePage {

    private static final String FIRST_NAME    = "#first-name";
    private static final String LAST_NAME     = "#last-name";
    private static final String POSTAL_CODE   = "#postal-code";
    private static final String CONTINUE_BTN  = "#continue";
    private static final String FINISH_BTN    = "#finish";
    private static final String CONFIRM_HDR   = ".complete-header";

    public CheckoutPage(Page page) {
        super(page);
    }

    public CheckoutPage fillShippingInfo(String firstName, String lastName, String postalCode) {
        waitForVisible(FIRST_NAME);
        locate(FIRST_NAME).fill(firstName);
        locate(LAST_NAME).fill(lastName);
        locate(POSTAL_CODE).fill(postalCode);
        locate(CONTINUE_BTN).click();
        return this;
    }

    public CheckoutPage finishOrder() {
        waitForVisible(FINISH_BTN);
        locate(FINISH_BTN).click();
        return this;
    }

    public boolean isOrderConfirmed() {
        waitForVisible(CONFIRM_HDR);
        return locate(CONFIRM_HDR).textContent().contains("Thank you");
    }
}
